libarchive/archive_read_support_format_rar.c

```diff
@@ -910,7 +910,7 @@ archive_read_format_rar_read_header(struct archive_read *a,
   const void *h;
   const char *p;
   struct rar *rar;
-  size_t skip;
+  int64_t skip;
```

Comment:
For consistency with 64-bit platforms, shouldn't this be `uint64_t`? If the risk of unsigned overflow only existed with a 32-bit `size_t`, it should be sufficient and perhaps more correct (since rar can't seek backwards(?)).

Reply:
Function prototypes were `off_t` many years ago and turned into `int64_t` while size of memory objects are, of course, `size_t`. So I stick with the types usually to be encountered here, i.e. `int64_t` and `size_t`. This leads to these ugly casts, but we switch between signed and unsigned implicitly anyway.